src/output.rs

```diff
@@ -639,7 +639,7 @@ where
             let mut destroyed = vec![];
```

Comment:
I think this should be an `Option`

Reply:
Hmm yeah that is a bit odd, only one surface can ever be destroyed at a time.

Instead of an `Option`, could we call `data.output_destroyed(conn, qh, output);` within the `retain` closure?

Reply:
No, there are borrow errors because `output_destroyed` takes `&mut self`.

Reply:
But it can be written without `retain`:

```rust
        if interface == "wl_output" {
            let output = data
                .output_state()
                .outputs
                .iter()
                .position(|o| o.name == name)
                .expect("Removed non-existing output");

            let wl_output = data.output_state().outputs[output].wl_output.clone();
            data.output_destroyed(conn, qh, wl_output);

            let output = data.output_state().outputs.remove(output);
            if let Some(xdg_output) = &output.xdg_output {
                xdg_output.destroy();
            }
            output.wl_output.release();
        }
```

Reply:
Yeah that would be better.